### DISCUSSION TITLE
Custom Module integration with other's SPA frameworks

### DISCUSSION THREAD

**Discussion by mpeix** [oqtane/oqtane.framework#2722] [Q&A]
Hi octane community,

We are thinking into integrate oqtane as part as our tech solutions. Currently we have most of the work done in DNN, but relying into React to build the UI and client side logic.

I have been playing a little bit with the solution created by oqtane Module Creator, trying to find if it is possible to integrate React into Razor  (That will allow us to work with a known technology and use all react components available).

What do you suggest as starting point?

Thank you

**Comment by sbwalker** [comment]
The main premise behind Blazor is that you can write full-stack applications using C#, without the need for JavaScript. As a result Blazor does not make it convenient to use JavaScript - it must be used through JS Interop which is different than what most frontend developers are accustomed to. Since Oqtane was built on Blazor I believe you will encounter some challenges.

**Comment by ADefWebserver** [comment]
Using React with Blazor (this is what Oqtane is written in) would be a bit like using React with Angular. With the same difficulties.

**Comment by mpeix** [comment]
Hi @sbwalker, @ADefWebserver,

thank you for your answers.

**Reply by sbwalker** [reply]
Microsoft did include support for "custom elements" in .NET 7 (https://www.telerik.com/blogs/embed-blazor-components-any-webpage-dotnet-7-official-custom-elements-support) but the use case is opposite of what you are describing - as it allows you to use Blazor UI components in your React application, whereas you want to use React UI components in a Blazor application.
